src/components/Menu/ActionMenu/ActionMenu.tsx

```diff
@@ -0,0 +1,75 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useMenu, useMenuItem, useMenuSection } from "@react-aria/menu";
+import { useTreeState } from "@react-stately/tree";
+import { ReactElement, useRef } from "react";
+import { AriaList } from "../Aria/AriaList";
+import { AriaMenuItem } from "../Aria/AriaMenuItem";
+import { AriaSection } from "../Aria/AriaSection";
+import { getKeyItemRecord, getMenuItems, mapToAriaProps } from "../Aria/helper";
+import { MenuItemType } from "../SelectMenu";
+
+export type ActionMenuItemType = MenuItemType & { onClick: () => void };
+
+export type ActionMenuBlock = {
+    id: string;
+    menuItems: ActionMenuItemType[];
+};
+
+export type ActionMenuProps = {
+    ariaLabel?: string;
+    menuBlocks: ActionMenuBlock[];
+};
+
+export const ActionMenu = ({
+    menuBlocks,
+    ariaLabel = "Action Menu",
+}: ActionMenuProps): ReactElement<ActionMenuProps> => {
+    const items = getMenuItems(menuBlocks);
+    const keyItemRecord = getKeyItemRecord(items);
+    const props = mapToAriaProps(ariaLabel, menuBlocks);
+    const state = useTreeState({
+        ...props,
+        selectionMode: "none",
+        disabledKeys: items.filter(({ disabled }) => disabled).map(({ id }) => id),
+    });
+    const menuRef = useRef<HTMLUListElement | null>(null);
+    const { menuProps } = useMenu(props, state, menuRef);
+
+    return (
+        <AriaList ariaProps={menuProps} ref={menuRef}>
+            {[...state.collection].map((section) => {
+                const sectionKey = section.key;
+                const { itemProps, groupProps } = useMenuSection({ "aria-label": `${sectionKey}` });
```

Comment:
```suggestion
                const { itemProps, groupProps } = useMenuSection({ "aria-label":  sectionKey });
```

Reply:
same here, I get a type error because `sectionKey` could be a number